execution/gethexec/node.go

```diff
@@ -99,6 +103,9 @@ func ConfigAddOptions(prefix string, f *flag.FlagSet) {
 	f.Uint64(prefix+".tx-lookup-limit", ConfigDefault.TxLookupLimit, "retain the ability to lookup transactions by hash for the past N blocks (0 = all blocks)")
 	f.Bool(prefix+".enable-prefetch-block", ConfigDefault.EnablePrefetchBlock, "enable prefetching of blocks")
 	StylusTargetConfigAddOptions(prefix+".stylus-target", f)
+	f.Int(prefix+".block-metadata-api-cache-size", ConfigDefault.BlockMetadataApiCacheSize, "size of lru cache storing the blockMetadata to service arb_getRawBlockMetadata.\n"+
+		"Note: setting a non-zero value would mean the blockMetadata might be outdated (if the block was reorged out).\n"+
```

Comment:
Need to think about this. 

I'm considering using execEngine to communicate with `BulkBlockMetadataFetcher` via channel whenever there's a reorg to signal clearing the cache

Reply:
I've implemented this, such that cache is cleared whenever there's a reorg. And I've chosen the direction of simply clearing the entire cache instead of target clearing of effected messages because I think Its not worth it to iterate over the cache to find which blocks were affected and clear one-by-one and all while holding a mutex.